### DISCUSSION TITLE
Version 7.0 takes longer than version 6.0 and has lower quality

### DISCUSSION THREAD

**Discussion by amhoho** [ImageMagick/ImageMagick#2011] [General]
### Result:
Version 7.0 costs twice as much as version 6.0

### Compare version:
Version: ImageMagick `7.0.10-6 Q16` x86_64 2020-04-17
Version: ImageMagick `6.7.8-9 Q16` 2019-08-08

### Test code:
```
cd /img/ && time -p maconvert ./bg/32.jpg ./block/base/9.png -alpha on -compose copy_opacity -geometry +185+57 -composite -crop 122X122+185+57 -sharpen 15 -compose over ./shadow/9.png -composite ./temp/000.block.png
```

### V6.0 time:
real 0.16 user 0.19 sys 0.00

### v7.0 time:
real 0.30 user 0.41 sys 0.00

### V6.0 Picture quality :
![03](https://user-images.githubusercontent.com/24509574/81758934-d0d8f800-94f5-11ea-8a29-df2f31f30e15.png)

**Comment by snibgo** [comment]
What does "-version" say for both versions? I suspect your v6 is not HDRI (ie uses integer arithmetic) where your v7 is HDRI (uses floating-point arithmetic). This would be the major difference in speed, although v7 is slightly slower than v6 as there is channel access has an extra level of indirection.

V6.7.8-9 is very very old. I think it had problems with linear RGB versus non-linear sRGB, which might account for the visual difference.

Please post your input files.

I suggest you add "+repage" before writing the output PNG.

**Reply by amhoho** [reply]
I also thought that slowness is caused by HDRI,so I recompile it with `./configure --disable-HDRI`,the result is still the same.The test picture is web picture found by Google.

**Comment by fmw42** [comment]
IM 6.7.8.9 has bugs. Also your IM 7 is not current. Try upgrading and keep it current. Also is your PNG delegate the same and even current on IM 7?

**Reply by amhoho** [reply]
I have updated to the latest version /ImageMagick-7.0.10-11 ,but the result is the same.

**Comment by fmw42** [comment]
IM 7 and IM 6 are now significantly different. IM 7 has feature enhancements. Some of which are very useful, but may slow IM 7 relative to the older IM 6. Variations in the JPG and PNG delegates may also slow things down. One of the IM developers will need to investigate the speed issue. But as I asked before, are you using the very same versions of JPG and PNG delegates. Those may have changed and cause differences in speed, especially for compressing the output PNG. Note that your IM 7 file size is smaller and so more compressed than your IM 6 file.

**Comment by snibgo** [comment]
> The test picture is web picture found by Google.

That doesn't help us try to reproduce the results. Please post the input images.
